test/Interpreter/keypath.swift

```diff
@@ -43,6 +73,20 @@ public func generic_class_constrained_keypath<U, V>(_ c: V) where V : GenericCon
   print(c[keyPath: kp].text)
 }
 
-// CHECK: Swift.KeyPath<main.GenericController<Swift.Int>, main.MyLabel>
+// CHECK: GenericController<Int>.label
 // CHECK: label
 generic_class_constrained_keypath(GenericController(5))
+
+// CHECK: {{\\Controller\.secondLabel!\.text|\\Controller\.<computed 0x.* \(Optional<MyLabel>\)>!\.<computed 0x.* \(String\)>}}
+print(\Controller.secondLabel!.text)
+
+// CHECK: {{\\Controller\.subscript\(_: String\)|\\Controller\.<computed 0x.* \(String\)>}}
+print(\Controller["abc"])
+// CHECK: \S.a
+print(\S.a)
+// CHECK: {{\\Controller\.subscript\(int: Int str: String _: Int\)|\\Controller\.<computed 0x.* \(Int\)>}}
+print(\Controller[int: 0, str: "", 0])
+// CHECK: {{\\Controller\.thirdLabel|\\Controller\.<computed 0x.* \(Optional<MyLabel>\)>}}
+print(\Controller.thirdLabel)
+// CHECK: {{\\Controller\.subscript\(\)|\\Controller\.<computed 0x.* \(Optional<Int>\)>}}
```

Comment:
Linux CI failure:

```
Input was:
<<<<<<
          1: label 
          2: \GenericController<Int>.label 
          3: label 
          4: \Controller.<computed 0x000055dafc924670 (Optional<MyLabel>)>!.<computed 0x000055dafc924720 (String)> 
          5: \Controller.<computed 0x000055dafc9248e0 (String)> 
          6: \S.a 
          7: \Controller.<computed 0x000055dafc924c00 (Int)> 
          8: \Controller.<computed 0x000055dafc924dc0 (Optional<MyLabel>)> 
check:91                                                                  X error: no match found
          9: \Controller.<computed 0x000055dafc924e70 (Int)> 
check:91     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>
```

It does look like your subscript is supposed to return `Int`, not `Optional<Int>`.

Reply:
I'll check into this later today. Not sure how this happened; I did test this on MacOS by turning off the happy-path for image lookup. But hopefully it will be a very simple fix.